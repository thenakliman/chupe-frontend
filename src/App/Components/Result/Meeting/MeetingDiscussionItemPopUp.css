.meeting-discussion-item-pop-up-container {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: rgba(0, 0, 0, 0.5);
}

.meeting-discussion-item-pop-up-inner {
  position: absolute;
  left: 20rem;
  top: 10rem;
  margin: auto;
  background: white;
  padding: 3rem;
  display: flex;
  flex-direction: column;
}

.button-container-meeting-discussion-item {
  display: flex;
  flex-direction: row;
  margin-bottom: 2rem;
}

.meeting-discussion-item-buttons {
  padding: 0.8rem 1.2rem;
  margin-right: 5rem;
  font-size: 1rem;
  cursor: pointer;
}

.meeting-discussion-item {
  border: 1px solid rgb(180, 180, 180);
  width: 45rem;
  height: 10rem;
  word-wrap: break-word;
  padding: 0.5rem;
  margin-bottom: 2rem;
}

.select-meeting-discussion-item-pop-up {
  padding: 0.6rem 1.5rem;
  width: 15rem;
  margin-bottom: 2rem;
}