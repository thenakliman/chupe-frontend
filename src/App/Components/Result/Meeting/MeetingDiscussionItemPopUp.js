import React from 'react';
import propTypes from 'prop-types';

require('./MeetingDiscussionItemPopUp.css');

export class MeetingDiscussionItemPopUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {discussionItem: '', discussionItemType: '', assignedTo: ''};
    this.handleSubjectChange = this.handleSubjectChange.bind(this);
    this.handleMeetingDiscussionItemTypeChange = this.handleMeetingDiscussionItemTypeChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  handleSubjectChange(value) {
    this.setState({discussionItem: value});
  }

  handleMeetingDiscussionItemTypeChange(value) {
    this.setState({discussionItemType: value});
  }

  handleAssignedToUserChange(value) {
    this.setState({assignedTo: value});
  }

  onSubmit() {
    this.props.onSubmit({
        discussionItem: this.state.discussionItem,
        discussionItemType: this.state.discussionItemType,
        assignedTo: this.state.assignedTo,
    });
  }

  render() {
    return (
      <div className={'meeting-discussion-item-pop-up-container'}>
      <div className={'meeting-discussion-item-pop-up-inner'}>
        <div>
          Meeting Discussion Item:
        </div>
          <textarea className={'meeting-discussion-item'}
                  value={this.state.discussionItem}
                  onChange={(event) => this.handleSubjectChange(event.target.value)}
                  resize={'none'}/>

        Assigned To
        <select value={this.state.assignedTo}
                id={'select-assigned-to'}
                className={'select-meeting-discussion-item-pop-up'}
                onChange={(event) => this.handleAssignedToUserChange(event.target.value)}
        >
        <option key={''} value={''}> Select user </option>
        {
          this.props.users.map((user) =>
            <option key={user.userName} value={user.userName}>
              {user.userName}
            </option>)
        }
        </select>
        Meeting Discussion Item Type:
        <select value={this.state.discussionItemType}
                id={'select-discussion-item-type'}
                className={'select-meeting-discussion-item-pop-up'}
                onChange={(event) => this.handleMeetingDiscussionItemTypeChange(event.target.value)}
        >
        <option key={''} value={''}> Select Discussion Item type </option>
        <option key={'INFORMATION'} value={'INFORMATION'}> Information </option>
        <option key={'ACTION_ITEM'} value={'ACTION_ITEM'}> Action Item </option>
        </select>
        <div className={'button-container-meeting-discussion-item'}>
          <button className={'meeting-discussion-item-buttons'}
                  id={'meeting-discussion-item-submit-button'}
                  onClick={this.onSubmit}>
              Submit
          </button>
          <button className={'meeting-discussion-item-buttons'}
                  id={'meeting-discussion-item-cancel-button'} onClick={this.props.onCancel}>
              Cancel
          </button>
        </div>
      </div>
      </div>
    );
  }
}

MeetingDiscussionItemPopUp.propTypes = {
  users: propTypes.array.isRequired,
  onSubmit: propTypes.func.isRequired,
  onCancel: propTypes.func.isRequired,
};
