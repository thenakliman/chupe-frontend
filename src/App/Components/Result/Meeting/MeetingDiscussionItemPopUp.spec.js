/* eslint-disable */
import React from 'react';
import {MeetingDiscussionItemPopUp} from './MeetingDiscussionItemPopUp';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';


const users = [{
  userName: 'user - 1',
}, {
  userName: 'user - 2',
}];

describe('Show meeting discussion item pop up component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have default discussion item empty string', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    expect(wrapper.state().discussionItem).toEqual('');
  });

  it('should have default users empty string', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    expect(wrapper.state().discussionItemType).toEqual('');
  });

  it('should set state to newly changed description', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    wrapper.find('textarea').simulate('change',
        {target: {value: 'description-value'}});

    expect(wrapper.state().discussionItem).toEqual('description-value');
  });

  it('should have option for each user and default for select', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    expect(wrapper.find('#select-assigned-to').find('option').length).toBe(3);
  });

  it('should set user on change of user select', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    wrapper.find('#select-assigned-to').simulate('change', {target: {value: 'user-1'}});

    expect(wrapper.state().assignedTo).toBe('user-1');
  });


  it('should have two option for discussion item type', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    expect(wrapper.find('#select-discussion-item-type').find('option').length).toBe(3);
  });

  it('should set discussion item type on change of type select', () => {
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={jest.fn()}
        />);

    wrapper.find('#select-discussion-item-type').simulate('change', {target: {value: 'ACTION_ITEM'}});

    expect(wrapper.state().discussionItemType).toBe('ACTION_ITEM');
  });

  it('should call on submit button', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={onSubmit}
            onCancel={jest.fn()}
        />);

    wrapper.setState({discussionItem: 'fake', discussionItemType: 'user'});

    wrapper.find('#meeting-discussion-item-submit-button').simulate('click');
    expect(onSubmit).toHaveBeenCalledWith({
      discussionItem: 'fake',
      discussionItemType: 'user',
      assignedTo: '',
    });
  });

  it('should call on cancel button', () => {
    const onCancel = jest.fn();
    const wrapper = shallow(
        <MeetingDiscussionItemPopUp
            users={users}
            onSubmit={jest.fn()}
            onCancel={onCancel}
        />);

    const feedback = {
      discussionItem: 'fake',
      discussionItemType: 'user-1',
    };

    wrapper.setState(feedback);

    wrapper.find('#meeting-discussion-item-cancel-button').simulate('click');
    expect(onCancel).toHaveBeenCalledWith();
  });
});
