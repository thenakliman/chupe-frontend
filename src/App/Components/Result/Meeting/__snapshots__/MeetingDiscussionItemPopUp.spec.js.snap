// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show meeting discussion item pop up component should match snapshot 1`] = `
<div
  className="meeting-discussion-item-pop-up-container"
>
  <div
    className="meeting-discussion-item-pop-up-inner"
  >
    <div>
      Meeting Discussion Item:
    </div>
    <textarea
      className="meeting-discussion-item"
      onChange={[Function]}
      resize="none"
      value=""
    />
    Assigned To
    <select
      className="select-meeting-discussion-item-pop-up"
      id="select-assigned-to"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
         Select user
      </option>
      <option
        key="user - 1"
        value="user - 1"
      >
        user - 1
      </option>
      <option
        key="user - 2"
        value="user - 2"
      >
        user - 2
      </option>
    </select>
    Meeting Discussion Item Type:
    <select
      className="select-meeting-discussion-item-pop-up"
      id="select-discussion-item-type"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
         Select Discussion Item type
      </option>
      <option
        key="INFORMATION"
        value="INFORMATION"
      >
         Information
      </option>
      <option
        key="ACTION_ITEM"
        value="ACTION_ITEM"
      >
         Action Item
      </option>
    </select>
    <div
      className="button-container-meeting-discussion-item"
    >
      <button
        className="meeting-discussion-item-buttons"
        id="meeting-discussion-item-submit-button"
        onClick={[Function]}
      >
        Submit
      </button>
      <button
        className="meeting-discussion-item-buttons"
        id="meeting-discussion-item-cancel-button"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
